main.go

```diff
@@ -132,3 +146,54 @@ func printThemes(cmd *cobra.Command) {
 		cmd.Println()
 	}
 }
+
+type numberWriter struct {
+	w           io.Writer
+	currentLine int64
+	buf         []byte
+}
+
+func (w *numberWriter) Write(p []byte) (n int, err error) {
+
+	var (
+		original = p
+		tokenLen int
+	)
+	for i, c := range original {
+		tokenLen++
+		if c != '\n' {
+			continue
+		}
+
+		token := p[:tokenLen]
+		p = original[i+1:]
+		tokenLen = 0
+
+		format := "%6d\t%s%s"
+		if w.currentLine > 999999 {
+			format = "%d\t%s%s"
+		}
+
+		_, er := fmt.Fprintf(w.w, format, w.currentLine, string(w.buf), string(token))
+		if er != nil {
+			return i + 1, er
+		}
+		w.buf = w.buf[:0]
+		w.currentLine++
+	}
+
+	if len(p) > 0 {
+		w.buf = append(w.buf, p...)
+	}
+	return len(original), nil
+}
+
+func (w *numberWriter) Flush() error {
+	format := "%6d\t%s"
+	if w.currentLine > 999999 {
+		format = "%d\t%s"
+	}
+	_, err := fmt.Fprintf(w.w, format, w.currentLine, string(w.buf))
```

Comment:
[ask]Can you remove the newline at EOF ❓ 

<img width="234" alt="toshimaru_toshimbp___src_github_com_toshimaru_nyan" src="https://user-images.githubusercontent.com/803398/96333179-53687380-10a3-11eb-8db6-d304ebf8fce6.png">

Reply:
This file is go.mod?
I checked the file, but the file ends with a newline to begin with. (i.e. the data entered has an EOF newline in the first place)
I do not think this should be trimmed on its own.

Reply:
Oh, there are no EOF line breaks in the file, that's my mistake in checking. I'll think of a solution.